functime/backtesting.py

```diff
@@ -112,6 +112,11 @@ def backtest(
     X: Optional[pl.DataFrame] = None,
     residualize: bool = True,
 ) -> Tuple[pl.DataFrame, pl.DataFrame]:
+    if forecaster.lags is None and X is None:
+        raise ValueError(
+            "When `lags` is None, `X` must be provided since no other features are available."
+        )
+
```

Comment:
I think we don't have to raise if X is None: the forecasting pipeline might include some feature extraction steps.

Reply:
Good point, I removed the check